### PR TITLE
Feat: add telemetry package and event (EventBuildOptimized)

### PR DESCRIPTION
**Pull request by harrytothemoon**
**What kind of change does this PR introduce?**

**Did you add tests(unit,e2e) for your changes?**

**Does this PR introduce a breaking change?**

**What needs to be documented once your changes are merged?**

### CODE REVIEW

File: packages/telemetry/src/helper/anonymous-meta.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    // Machine information
+    cpuCount: cpus.length,
+    cpuModel: cpus.length ? cpus[0].model : null,
+    cpuSpeed: cpus.length ? cpus[0].speed : null,
+    memoryInMb: Math.trunc(os.totalmem() / Math.pow(1024, 2)),
+    // Environment information
+    isDocker: isDockerFunction(),
+    isWsl: isWslBoolean,
+    isCI: ciEnvironment.isCI,
+    ciName: (ciEnvironment.isCI && ciEnvironment.name) || null,
```
Comment (evenchange4) [inline packages/telemetry/src/helper/anonymous-meta.ts:44]: TBC
- [ ] make sure the `ci-info` support more ci, e.g., `flow-ci`.

File: packages/telemetry/src/telemetry.ts [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+      // `conf` incorrectly throws a permission error during initialization
+      // instead of waiting for first use. We need to handle it, otherwise the
+      // process may crash.
+      this.conf = new Conf({ projectName: 'shuvijs' });
+    } catch (_) {
+      this.conf = null;
+    }
+    this.sessionId = randomBytes(32).toString('hex');
+    this.rawProjectId = getRawProjectId();
+    this.postEndpoint = postEndpoint || 'TODO: update end point';
```
Comment (evenchange4) [inline packages/telemetry/src/telemetry.ts:74]: I will suggest if the `postEndpoint` option is not provided, just skip the `_postPayload`.

**Review by evenchange4 — CHANGES_REQUESTED** [review]

File: packages/service/src/bundler/bundler.ts [hunk 3]
```diff
@@ -130,6 +141,61 @@ class WebpackBundler implements Bundler {
     return this._compiler.compilers.find(compiler => compiler.name === name);
   }
 
+  async analysis() {
```
Comment (liximomo) [inline packages/service/src/bundler/bundler.ts:144]: 这个方法有啥必要放到bundler里？

File: packages/service/src/core/api.ts [hunk 4]
```diff
@@ -92,7 +97,9 @@ class Api {
     plugins,
     phase,
     platform,
-    normalizePlatformConfig
+    normalizePlatformConfig,
+    telemetry,
+    version
```
Comment (liximomo) [inline packages/service/src/core/api.ts:102]: version 这种不可变的东西，做成option不合理

**Review by liximomo — CHANGES_REQUESTED** [review]

File: packages/telemetry/src/telemetry.ts (hunk 2 above)
Comment (harrytothemoon) [reply packages/telemetry/src/telemetry.ts]: ok
https://github.com/shuvijs/shuvi/pull/473/commits/381a2a07b7c170dc41d49dda8bfc2fbee6a699c6

File: packages/service/src/bundler/bundler.ts (hunk 3 above)
Comment (harrytothemoon) [reply packages/service/src/bundler/bundler.ts]: refactor:
https://github.com/shuvijs/shuvi/pull/473/commits/b113ec7921327b78832b3fba3fe3bb649ac7d908

File: packages/service/src/core/api.ts (hunk 4 above)
Comment (harrytothemoon) [reply packages/service/src/core/api.ts]: ok

File: packages/service/src/core/api.ts [hunk 5]
```diff
@@ -111,7 +111,8 @@ class Api {
     this._pluginManager.clear();
     this._projectBuilder = new ProjectBuilder();
     this._normalizePlatformConfig = normalizePlatformConfig;
-    this._telemetry = telemetry || new Telemetry({});
+    this._telemetry =
```
Comment (liximomo) [inline packages/service/src/core/api.ts:114]: postEndpoint 是不是应该先设个空值

File: packages/telemetry/src/telemetry.ts [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+      this.conf = null;
+    }
+    this.sessionId = randomBytes(32).toString('hex');
+    this.rawProjectId = getRawProjectId();
+    this.postEndpoint = postEndpoint;
+
+    this.queue = new Set();
+
+    this.notify();
+    this.meta = { ...getAnonymousMeta(process.env._SHUVI_VERSION!), ...meta };
```
Comment (liximomo) [inline packages/telemetry/src/telemetry.ts:77]: 这是什么操作？不能从shuvi/service那边传进来吗

File: packages/shuvi/src/shuvi.ts [hunk 7]
```diff
@@ -19,7 +21,9 @@ export async function initShuvi({
   config,
   ...options
 }: ShuviOption): Promise<Api> {
+  process.env._SHUVI_VERSION = pkgInfo.version;
```
Comment (liximomo) [inline packages/shuvi/src/shuvi.ts:24]: 不要这样用env共享全局变量！

**Review by liximomo — CHANGES_REQUESTED** [review]

**Review by liximomo — APPROVED** [review]
/lgtm

**Pull request merged**